app/src/main/java/cm/aptoide/pt/abtesting/ABTestService.java

```diff
@@ -20,35 +21,37 @@
   private static final String EXPERIMENT_DRAFT = "EXPERIMENT_IN_DRAFT_STATE";
 
   private ServiceV7 service;
-  private String aptoideId;
+  private IdsRepository idsRepository;
 
-  public ABTestService(ServiceV7 service, String aptoideId) {
+  public ABTestService(ServiceV7 service, IdsRepository idsRepository) {
     this.service = service;
-    this.aptoideId = aptoideId;
+    this.idsRepository = idsRepository;
   }
 
   public Observable<ExperimentModel> getExperiment(String identifier) {
-    return service.getExperiment(identifier, aptoideId)
+    return getAptoideId().flatMap(aptoideId -> service.getExperiment(identifier, aptoideId)
```

Comment:
we are not making sure this (getAptoideId()) is not running in the main thread, this might be a problem